reactor-core-2/src/test/java/com/baeldung/reactor/flux/parallelflux/ParallelFluxUnitTest.java

```diff
@@ -0,0 +1,56 @@
+package com.baeldung.reactor.flux.parallelflux;
+
+import com.baeldung.reactor.fluxvsparallelflux.Fibonacci;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.ParallelFlux;
+import reactor.core.scheduler.Schedulers;
+import reactor.test.StepVerifier;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+
+@Slf4j
+public class ParallelFluxUnitTest {
+
+    @Test
+    public void givenFibonacciIndices_whenComputingWithParallelFlux_thenCorrectResults() {
+        AtomicLong timeTakenForCompletion = new AtomicLong();
+
+        ParallelFlux<Long> parallelFluxFibonacci = Flux.just(43, 44, 45, 47, 48)
+                .parallel(3)
+                .doOnSubscribe(subscription -> timeTakenForCompletion.set(-1 * System.nanoTime()))
+                .runOn(Schedulers.parallel())
+                .map(Fibonacci::fibonacci);
+
+        Flux<Long> sequencialParallelFlux = parallelFluxFibonacci.sequential()
+                .doFinally(consumer -> timeTakenForCompletion.addAndGet(System.nanoTime()));
+
+        StepVerifier.create(sequencialParallelFlux)
+                .expectNextMatches(n -> List.of(433494437L, 701408733L, 1134903170L, 2971215073L, 4807526976L).contains(n))
+                .expectNextMatches(n -> List.of(433494437L, 701408733L, 1134903170L, 2971215073L, 4807526976L).contains(n))
+                .expectNextMatches(n -> List.of(433494437L, 701408733L, 1134903170L, 2971215073L, 4807526976L).contains(n))
+                .expectNextMatches(n -> List.of(433494437L, 701408733L, 1134903170L, 2971215073L, 4807526976L).contains(n))
+                .expectNextMatches(n -> List.of(433494437L, 701408733L, 1134903170L, 2971215073L, 4807526976L).contains(n))
+                .verifyComplete();
+
+
+        log.info("Total time taken for Parallel Flux pipeline processing Fibonacci indices 43, 44, 45, 47, 48: {} ms", Duration.ofNanos(timeTakenForCompletion.get()).toMillis());
+    }
+
+    @Test
+    public void givenListOfIds_whenComputingWithParallelFlux_OrderChanges() {
```

Comment:
Not sure what "OrderChanges" is supposed to mean in the "then" part of the method name?